File: src/api/schemas/para.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel


class ParaFileRead(BaseModel):
    id: str
    name: str
    relative_path: str
    last_modified_at: datetime


class ParaProjectRead(BaseModel):
    id: str
    name: str
    path: str
    file_count: int
    last_modified_at: Optional[datetime] = None
    files: List[ParaFileRead]

File: front/lib/types/para.ts
export interface ParaFileRead {
  id: string;
  name: string;
  relative_path: string;
  last_modified_at: string;
}

export interface ParaProjectRead {
  id: string;
  name: string;
  path: string;
  file_count: number;
  last_modified_at: string | null;
  files: ParaFileRead[];
}
